### PR TITLE
Add default status for new user

### LINKED ISSUE #766: Add default status for new user
### AC
Add default status idle when a new user is created

### Description:
When a user joins RDS, the default status of the user is "idle". To represent this,  in the user modal we have `status: "idle"`. This field can also be used as a query in firestore to filter the idle users https://github.com/Real-Dev-Squad/website-backend/issues/635

### PR DESCRIPTION
**Pull request by SanketDhabarde**
### Changes made
Updated `addOrUpdate` model function to add the default status for new user (i.e. "idle")

### The order of PR
https://github.com/Real-Dev-Squad/website-backend/pull/738#issuecomment-1297216403

### CODE REVIEW

File: models/users.js [hunk 1]
```diff
@@ -49,6 +50,17 @@ const addOrUpdate = async (userData, userId = null) => {
...
     userData.incompleteUserDetails = true;
+    /**
+     * Adding default status to user enable to query for idle user
+     * This can be used in /users/idle
+     * https://github.com/Real-Dev-Squad/website-backend/issues/635
+     * The reason I am checking whether status exists or not before adding
+     * 1. If we are adding the user with some status it will not get override
+     * 2. Some tests are depends on the status of user, We are adding the user before each tests, those will consider as new user only and the status will be override if we don't check
+     */
+    if (userData.status === undefined) {
```
Comment (samarpan1738) [inline models/users.js:61]: Ideally we should not write code to pass the test cases. 
If test cases are breaking we can edit them

Comment (SanketDhabarde) [reply models/users.js]: Thank you for pointing out. Made the change. please check @samarpan1738

File: test/integration/tasks.test.js [hunk 2]
```diff
@@ -30,6 +31,8 @@ describe("Tasks", function () {
     const superUserId = await addUser(superUser);
     jwt = authService.generateAuthToken({ userId });
     superUserJwt = authService.generateAuthToken({ userId: superUserId });
+    // Adding status to active user as while adding the user it will be consider as new user (for new user default status is idle)
```
Comment (samarpan1738) [inline test/integration/tasks.test.js:34]: Is this required? Feels redundant to me

Comment (SanketDhabarde) [reply test/integration/tasks.test.js:34]: After adding the user which already has status, why I am adding the status again. May feel confused at first. That's why I added comment

**Review by samarpan1738 — APPROVED** [review]
LGTM

File: models/users.js [hunk 3]
```diff
@@ -49,6 +50,12 @@ const addOrUpdate = async (userData, userId = null) => {
      */
     userData.roles = { archived: false };
     userData.incompleteUserDetails = true;
+    /**
+     * Adding default status to user enable to query for idle user
+     * This can be used in /users/idle
+     * https://github.com/Real-Dev-Squad/website-backend/issues/635
```
Comment (ivinayakg) [inline models/users.js:56]: you don't need to mention this.

**Review by ivinayakg — APPROVED** [review]
LGTM

File: models/users.js (hunk 3 above)
Comment (SanketDhabarde) [reply models/users.js:56]: Ok will remove. Thank you.

**Review by vinayak-trivedi — DISMISSED** [review]
LGTM 👍

**Review by ivinayakg — APPROVED** [review]

**Review by vinayak-trivedi — APPROVED** [review]

**Review by harshith-venkatesh — COMMENTED** [review]
Are there no test cases that needs to be updated for the changes here?

**Comment by SanketDhabarde** [comment]
> Are there no test cases that needs to be updated for the changes here?

I was not able to think of any use case. Also I haven't seen in codebase that we are testing the property in user modal. The test case will come in next pr https://github.com/Real-Dev-Squad/website-backend/pull/738#issuecomment-1297216403
If you have any suggestions please let me know.

**Comment by SanketDhabarde** [comment]
Closing this PR. As the new status collection is getting created which make this issue irrelevant https://github.com/Real-Dev-Squad/website-backend/issues/818

**Pull request closed without merging**